packages/grid_client/tests/modules/gateways.test.ts

```diff
@@ -18,6 +26,42 @@ beforeAll(async () => {
   return gridClient;
 });
 
+async function waitForGateway(domain: string) {
+  for (let i = 0; i < 30; i++) {
+    try {
+      await axios.get(domain);
+      log("gateway is reachable");
+      return true;
+    } catch (error) {
+      log("gateway is not reachable");
+    }
+
+    const wait = await setTimeout(5000, "Waiting for gateway to be ready");
+    log(wait);
```

Comment:
I replaced the wait in the catch to make sure it had the correct effect, but I couldn't remove the loop since if the gateway failed, it would run forever.